### ISSUE TITLE
estimate_homography(refine=False) ignores model="sRT" — _ransac_init_one is always 8-DOF

### ISSUE THREAD

**Issue by macnack** [macnack/ransac_multimodel#3]
## Summary

`estimate_homography(..., refine=False)` silently ignores its own `model` parameter and always
returns an 8-DOF full homography, even when `model="sRT"` (the function's default). The same
bug, independently, exists in `homography_theseus.py`'s `_ransac_init`, used by the `refine=True`
theseus/torch-LM path.

## Where

`pipeline.py`:

- `estimate_homography(logits, *, backend="numpy", model="sRT", ..., refine=True, ...)` — the
  `refine=False` branch (`backend="numpy"`) calls:
  ```python
  H_init = _ransac_init_one(
      pts_A, peaks_B, means_B, use_means_for_ransac,
      ransac_method, ransac_reproj_threshold,
      ransac_max_iters, ransac_confidence,
      init_backend=init_backend,
  )
  H = H_init
  ```
  `model` is never passed.
- `_ransac_init_one(pts_A, peaks_B, means_B, use_means, ransac_method, reproj, max_iters,
  confidence, init_backend="cv2", device="cpu")` has **no `model` parameter at all**, and its
  `init_backend="cv2"` branch unconditionally calls:
  ```python
  H, mask = cv2.findHomography(pts_A, target, ransac_method, ...)
  ```
  — always 8-DOF, never `cv2.estimateAffinePartial2D`.
- The same pattern exists in `homography_theseus.py`'s `_ransac_init` (no `model` param, always
  `cv2.findHomography`), used by the theseus-backed `refine=True` path.

Confirmed on `main` (commit `023165d`, the current HEAD of `pipeline.py`).

## Impact

Any caller that relies on `estimate_homography`'s documented/default `model="sRT"` behavior with
`refine=False` silently gets an 8-DOF homography instead of a 4-DOF similarity — no error, no
warning, wrong geometry. Confirmed live (not just theoretical) via a downstream consumer:
`sat_roma`'s `experiments/lpc/tier0_sigma_oracle_gate.py` calls

```python
estimate_homography(gm_cls, backend="numpy", refine=False, use_means_for_ransac=False)
```

without overriding `model` — every result it has ever produced is an 8-DOF fit, not the sRT fit
its own defaults (and presumably its analysis) assume.

Note: `sat_roma`'s *other* consumer of this package,
`romatch/eval/multimodel_homography.py::estimate_multimodel_homography`, works around this
correctly today by bypassing `estimate_homography` entirely on `refine=False` and calling
`cv2.estimateAffinePartial2D` directly for `model="sRT"` — so the paper/production headline
numbers are unaffected. This issue is about `estimate_homography` itself misbehaving for any
other/future caller who reasonably trusts its own `model` parameter.

## Suggested fix

Thread `model` into `_ransac_init_one` / `_ransac_init` (default `"sRT"` to match
`estimate_homography`'s own default) and dispatch on it in the `init_backend="cv2"` branch:

```python
if model == "sRT":
    M, mask = cv2.estimateAffinePartial2D(pts_A, target, method=ransac_method,
                                          ransacReprojThreshold=reproj, maxIters=max_iters,
                                          confidence=confidence)
    H = None if M is None else np.vstack([M, [0.0, 0.0, 1.0]])
else:
    H, mask = cv2.findHomography(pts_A, target, ransac_method, ...)
```

Happy to send a PR with this if useful — flagging first since it changes numerical output for
existing `refine=False, model="sRT"` callers.

**Comment by macnack** [comment]
Same root cause reaches two more call sites in `estimate_homography_batched`:

- `pipeline.py:918-932` — both `_ransac_init_one` calls in the `ransac_sampling == "uniform"`
  branch (with and without `refine_inliers_only`) omit `model`, same as the single-frame path.
- `pipeline.py:934-940` — the `else` branch (`ransac_sampling in ("p_mode", "p_mode_power")`)
  unconditionally calls `_ransac_init_weighted` (DLT/homography-only, no `model` concept), never
  `_ransac_init_weighted_srt` (the sRT-capable sibling — see #4, currently unreachable from any
  public API for the same reason).
- The GPU-native backends (`_batched_init_from_gpu`, used for `init_backend in ("kornia_dlt",
  "kornia_dlt_iter", "torch_ransac")`) don't accept a `model` parameter at all — homography-only
  regardless of what the caller requested. Fixing this needs a similarity-DLT implementation
  (kornia has no built-in one); worth scoping as a separate follow-up rather than folding into
  the immediate fix here.

Also worth noting while fixing this: there are three independent single-frame RANSAC-init
implementations in the package — `ransac_init.py::ransac_init` (correct, already parameterized
by `estimator="homography"|"similarity"`, and what `romatch/eval/multimodel_homography.py` in
the downstream sat_roma repo uses directly to work around this whole issue) vs. `pipeline.py`'s
`_ransac_init_one` and `homography_theseus.py`'s `_ransac_init` (both missing `model`). The
cleanest fix might be deleting the two duplicates in favor of the one that's already correct,
rather than adding a third `model` branch.
